### ISSUE TITLE
Missing values in time columns not written correctly

### ISSUE THREAD

**Issue by J535D165** [tidyverse/readr#715]
Init
``` R
> x <- tibble(
    time = parse_time(c("11:10:09", "12:11:10", NA)), 
    date = parse_date(c("1990-12-12", NA, "1990-12-12")), 
    int = c(NA, 10, 12))
> x
# A tibble: 3 x 3
      time       date   int
    <time>     <date> <dbl>
1 11:10:09 1990-12-12    NA
2 12:11:10         NA    10
3       NA 1990-12-12    12
```

Problem. The NA values in the time column are handled differently.  

``` 
write_delim(x, "test.csv")

time,date,int
11:10:09,1990-12-12,NA
12:11:10,NA,10
"NA",1990-12-12,12
```

```
write_delim(x, "test.csv", na="")

time,date,int
11:10:09,1990-12-12,
12:11:10,,10
NA,1990-12-12,12

```

Also calling View(x) in Rstudio shows that there is something different with the time column (NA has slightly different colour). 

Btw, I hope that this is not a duplicate. 

Kind regards, Jonathan

**Comment by jimhester** [comment]
Simpler reprex is

``` r
library(readr)
format_csv(tibble::tibble(x = parse_time(NA)))
#> [1] "x\n\"NA\"\n"
```

Created on 2017-12-07 by the [reprex package](http://reprex.tidyverse.org) (v0.1.1.9000).

Output should be `"x\nNA\n"`

**Comment by jimhester** [comment]
This is because NA's are not preserved by hms, https://github.com/tidyverse/readr/issues/715 will fix this once it is merged.

**Comment by krlmlr** [comment]
Can this be closed with hms >= 0.4.1?
